mapiproxy/servers/default/emsmdb/dcesrv_exchange_emsmdb.c

```diff
@@ -1542,12 +1542,14 @@ static enum MAPISTATUS dcesrv_EcDoAsyncConnectEx(struct dcesrv_call_state *dce_c
 	struct dcesrv_handle		*handle;
 	struct dcesrv_endpoint		*item;
 	bool				found;
+	const char			*email;
 
 	OC_DEBUG(3, "exchange_emsmdb: EcDoAsyncConnectEx (0xe)\n");
 
 	/* Step 0. Ensure incoming user is authenticated */
 	if (!dcesrv_call_authenticated(dce_call)) {
 		OC_DEBUG(1, "No challenge requested by client, cannot authenticate");
+	failure:
```

Comment:
is easier to read the same 4 lines over and over again when there is an error on the same function? For me this is the best scenario and justification to use `goto` (and is used a lot of times on openchange code, `grep -rn "goto failure"`)